app/assets/javascripts/application.js.erb

```diff
@@ -515,16 +465,41 @@ const onPageLoad = (callback) => {
 const replace = (oldElement, newHTMLString) => {
   const placeHolder = document.createElement('div');
   placeHolder.insertAdjacentHTML('afterBegin', newHTMLString);
-  const newElement = placeHolder.firstElementChild;
-  oldElement.replaceWith(newElement);
+  const newElements = placeHolder.childNodes.values();
+  oldElement.replaceWith(...newElements);
+}
+
+const fetchWithCSRF = (url, params) => {
+  let csrfToken = document.querySelector('meta[name="csrf-token"]').getAttribute('content');
+  let h = new Headers();
+  h.append('X-CSRF-Token', csrfToken);
+  params = params || {};
+  params['headers'] = h;
+  return fetch(url, params);
+} 
+
+const removeExtraInput = (targetElement) => {
+  // This deletion will resovle a bug that creates an extra hidden input when rendering the comment elements.
+  const extraInput2 = targetElement.parentElement.querySelector('.comment_folder_button');
+  extraInput2.remove();
 }
 
 onPageLoad(() => {
 
+  // Global Functions
   on('click', '.markdown_help_label', (event) => {
     parentSelector(event.target, '.markdown_help_toggler').querySelector('.markdown_help').classList.toggle('display-block');
   });
 
+  // Inbox Related Funtions
+
+  on('change', '#message_hat_id', (event) => {
+    let selectedOption = event.target.options.selectedIndex;
+    if (event.target.options[selectedOption].getAttribute('data-modnote') == 'true') {
```

Comment:
Replace the `if`: `document.getElementById('message_mod_note').checked = event.target.options[selectedOption].getAttribute('data-modnote') === 'true');` (but see previous comment)

Reply:
That is a great refactoring.  Thanks for streamlining it.